requests/utils.py

```diff
@@ -87,7 +83,22 @@ def super_len(o):
             # is actually a special file descriptor like stdin. In this
             # instance, we don't know what the length is, so set it to zero and
             # let requests chunk it instead.
-            current_position = total_length
+            if total_length is not None:
+                current_position = total_length
+
+        if hasattr(o, 'seek') and total_length is None:
+            # StringIO and BytesIO have seek but no useable fileno
```

Comment:
I believe that's what this comment conveys @tzickel, they all have seek and no *useable* form of `fileno`. Was there something you were looking to see changed to make the comment clearer? 99364b6 was added to remove the use of seek to set `total_length` because of the inconsistency between the `io` and `StringIO` modules.